evalml/tests/component_tests/test_time_series_imputer.py

```diff
@@ -0,0 +1,478 @@
+import numpy as np
+import pandas as pd
+import pytest
+import woodwork as ww
+from pandas.testing import assert_frame_equal, assert_series_equal
+from woodwork.logical_types import (
+    Boolean,
+    Categorical,
+    Double,
+    Integer,
+    NaturalLanguage,
+)
+
+from evalml.pipelines.components import TimeSeriesImputer
+
+
+def test_invalid_strategy_parameters():
+    with pytest.raises(ValueError, match="Valid numeric impute strategies are"):
+        TimeSeriesImputer(numeric_impute_strategy="mean")
+    with pytest.raises(ValueError, match="Valid categorical impute strategies are"):
+        TimeSeriesImputer(categorical_impute_strategy="interpolate")
+    with pytest.raises(ValueError, match="Valid target column impute strategies are"):
+        TimeSeriesImputer(target_impute_strategy="not a strategy")
+
+
+def test_imputer_default_parameters():
+    imputer = TimeSeriesImputer()
+    expected_parameters = {
+        "categorical_impute_strategy": "forwards_fill",
+        "numeric_impute_strategy": "interpolate",
+        "target_impute_strategy": "forwards_fill",
+    }
+    assert imputer.parameters == expected_parameters
+
+
+@pytest.mark.parametrize(
+    "categorical_impute_strategy", ["forwards_fill", "backwards_fill"]
+)
+@pytest.mark.parametrize(
+    "numeric_impute_strategy", ["forwards_fill", "backwards_fill", "interpolate"]
+)
+@pytest.mark.parametrize(
+    "target_impute_strategy", ["forwards_fill", "backwards_fill", "interpolate"]
+)
+def test_imputer_init(
+    target_impute_strategy, numeric_impute_strategy, categorical_impute_strategy
+):
+
+    imputer = TimeSeriesImputer(
+        categorical_impute_strategy=categorical_impute_strategy,
+        numeric_impute_strategy=numeric_impute_strategy,
+        target_impute_strategy=target_impute_strategy,
+    )
+    expected_parameters = {
+        "categorical_impute_strategy": categorical_impute_strategy,
+        "numeric_impute_strategy": numeric_impute_strategy,
+        "target_impute_strategy": target_impute_strategy,
+    }
+    expected_hyperparameters = {
+        "categorical_impute_strategy": ["backwards_fill", "forwards_fill"],
+        "numeric_impute_strategy": ["backwards_fill", "forwards_fill", "interpolate"],
+        "target_impute_strategy": ["backwards_fill", "forwards_fill", "interpolate"],
+    }
+    assert imputer.name == "Time Series Imputer"
+    assert imputer.parameters == expected_parameters
+    assert imputer.hyperparameter_ranges == expected_hyperparameters
+
+
+def test_numeric_only_input(imputer_test_data):
```

Comment:
I think we should come up with a test case where forwards_fill, backwards_fill, interpolate yield different answers. Otherwise we could easily introduce a bug and not notice.

Reply:
This test does cover that scenario - the third value in the "float with nan" column is -1, 1, or 0 depending on whether the imputation strategy is backwards fill, forwards fill, or interpolate, respectively. It can be hard to notice though, so I am happy to add an explicit test if that makes more sense!

Reply:
🤦‍♂️